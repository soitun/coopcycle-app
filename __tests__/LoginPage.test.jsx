import React from 'react'
import LoginForm from '../src/components/LoginForm'
import { shallow } from 'enzyme'


it('renders correcty', () => {
  const tree = shallow(<LoginForm  />)
  expect(tree).toMatchSnapshot()
})

it('update state when entering input', () => {
  const tree = shallow(<LoginForm  />),
    usernameInput = tree.find('Styled(Input)').at(0),
    passwordInput = tree.find('Styled(Input)').at(1)

  usernameInput.props().onChangeText('Hello')
  passwordInput.props().onChangeText('secret')

  expect(tree.state()).toEqual({ email: 'Hello', password: 'secret', error: false })
})

it('submit', () => {

  const onRequestStart = jest.fn(),
    onRequestEnd = jest.fn(),
    onLoginSuccess = jest.fn(),
    onLoginFail = jest.fn(),
    login = jest.fn().mockReturnValueOnce(new Promise((resolve, reject) => { resolve() })),
    client = { login: login },
    tree = shallow(
      <LoginForm
        onRequestStart={onRequestStart}
        onRequestEnd={onRequestEnd}
        onLoginSuccess={onLoginSuccess}
        onLoginFail={onLoginFail}
        client={client}
      />),
    usernameInput = tree.find('Styled(Input)').at(0),
    passwordInput = tree.find('Styled(Input)').at(1),
    submitButton = tree.find('Styled(Button)').at(0)

  usernameInput.props().onChangeText('Hello')
  passwordInput.props().onChangeText('secret')
  submitButton.props().onPress()

  expect(tree.state()).toEqual({ email: 'Hello', password: 'secret', error: false })
  expect(login).toHaveBeenCalledTimes(1)
  expect(login).toHaveBeenCalledWith('Hello', 'secret')
})