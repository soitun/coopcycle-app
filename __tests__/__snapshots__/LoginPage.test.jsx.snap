// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correcty 1`] = `
<View>
  <Styled(Form)>
    <Styled(Item)
      stackedLabel={true}
    >
      <Styled(Label)>
        Nom d'utilisateur
      </Styled(Label)>
      <Styled(Input)
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        style={
          Object {
            "height": 40,
          }
        }
      />
    </Styled(Item)>
    <Styled(Item)
      stackedLabel={true}
    >
      <Styled(Label)>
        Mot de passe
      </Styled(Label)>
      <Styled(Input)
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        secureTextEntry={true}
        style={
          Object {
            "height": 40,
          }
        }
      />
    </Styled(Item)>
  </Styled(Form)>
  <View
    style={
      Object {
        "marginTop": 20,
        "paddingHorizontal": 10,
      }
    }
  >
    <Styled(Button)
      block={true}
      onPress={[Function]}
    >
      <Styled(Text)>
        Valider
      </Styled(Text)>
    </Styled(Button)>
  </View>
</View>
`;
